### DISCUSSION TITLE
Multinode OC CMS without any cloud service

### DISCUSSION THREAD

**Discussion by ns8482e** [OrchardCMS/OrchardCore#14318] [Q&A] [answered]
Is it possible to have multi-node setup without using any cloud service like azure blob?

**Comment by Piedone** [comment]
If you mean only the storage aspect then I guess some kind of shared folder would work too.

**Reply by ns8482e** [reply]
I tried with shared folder, when enabled a feature, worked on one node but when request sent to second node it returned 404

**Comment by jtkech** [comment, marked as answer]
With Redis features most of things can be shared without using any cloud service.

Unless for Media and Lucene (you may need ElasticSearch).

**Reply by ns8482e** [reply]
I tried with shared folder for same site but when setting updated the other node didn't reload the changes

**Reply by ns8482e** [reply]
@jtkech  if I use redis, is redis connection shared by all tenants or each tenant will create it's own connection? 
However Enabling redis connection at host level using following fixed multi-node inconistancy

```cs
services.AddOrchardCms()
	.AddGlobalFeatures("OrchardCore.Redis.Cache")
	.AddGlobalFeatures("OrchardCore.Redis.Lock")
```
